File: src/gestorAplicación/modelos/Cliente.java
package gestorAplicacion.modelos;

import java.io.Serializable;

import gestorAplicacion.operaciones.Venta;

public class Cliente extends Persona implements Serializable {
    private static final long serialVersionUID = 4L;

    public Cliente() {
        super();
    }

    public Cliente(String nombre, String documento, String email) {
        super(nombre, documento, email);
    }

    public double calcularFidelidad(int numeroVentasTienda) {
        return getHistorico().size() / numeroVentasTienda;
    }

    public void agregarCompra(Venta compra) {
        getHistorico().add(compra);
    }
}

File: python/gestorAplicacion/modelos/Cliente.py
from gestorAplicacion.modelos.Persona import Persona


class Cliente(Persona):
    def __init__(self, nombre, documento, email):
        super().__init__(nombre, documento, email)

    def calcularFidelidad(self, numeroVentasTienda):
        return 0 if (numeroVentasTienda == 0) else len(super().getHistorico())/numeroVentasTienda
